.eslintrc.js

```diff
@@ -0,0 +1,523 @@
+module.exports = {
+    env: {
+        browser: true,
+        es6: true
+    },
+    extends: [
+        "eslint:recommended",
+        "plugin:@typescript-eslint/eslint-recommended",
+        "plugin:@typescript-eslint/recommended",
+        "plugin:react/recommended",
+        "plugin:@typescript-eslint/recommended-requiring-type-checking",
+        "prettier",
+        "plugin:prettier/recommended"
+    ],
+    parser: "@typescript-eslint/parser",
+    parserOptions: {
+        project: "tsconfig.json",
+        sourceType: "module",
+        projectFolderIgnoreList: ["node_modules"],
+        createDefaultProgram: false,
+        tsconfigRootDir: `${__dirname}`,
+        ecmaFeatures: {
+            jsx: true // Allows for the parsing of JSX
+        }
+    },
+    plugins: ["@typescript-eslint", "react", "react-hooks"],
+    settings: {
+        react: {
+            createClass: "Component",
+            version: "detect"
+        }
+    },
+    rules: {
+        'react/react-in-jsx-scope': 'off', // TODO: discuss if we need to turn this on and address the issue
+        'react/no-unescaped-entities': 'off', // TODO: discuss if this should be turned on and address the issue
+        "@typescript-eslint/adjacent-overload-signatures": "error",
+        "@typescript-eslint/array-type": [
+            "error",
+            {
+                default: "array"
+            }
+        ],
+        "@typescript-eslint/await-thenable": "error",
+        "no-unused-vars": "off",
```

Comment:
why is this off?

Reply:
I believe this is set to off because it is replaced by the TypeScript-specific version of the rule, `@typescript-esline/no-unused-vars` in the following lines below which is currently throwing linting errors in multiple files in the project including App.test.tsx, Navbar.tsx, IconMyAShort.tsx, usingInDotnet, usintInJava, etc for unused variables.

It appears that turning off the base ESLint rule and using the TypeScript-enhanced version is used to ensure better compatibility with TypeScript codebases and more accurate linting results (avoid false positives, properly handles type information and can more accurately determine whether variables, functions, and imports are actually used or not)